SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -2151,211 +2151,141 @@ void CNEMOEulerSolver::BC_Outlet(CGeometry *geometry, CSolver **solver_container
+ const su2double* Velocity       = config->GetInlet_Velocity(Marker_Tag);
+ const su2double Temperature_ve = Temperature; // TODO : add/check for second function for Tve, check what inputs are in config
+
+ /*--- Set mixture state ---*/
+ FluidModel->SetTDStatePTTv(Pressure, Mass_Frac, Temperature, Temperature_ve);
+
+ /*--- Compute necessary quantities ---*/
+ const su2double rho = FluidModel->GetDensity();
+ const su2double soundspeed = FluidModel->ComputeSoundSpeed();
+ const su2double sqvel = GeometryToolbox::SquaredNorm(nDim, Velocity);
+ const auto& energies = FluidModel->ComputeMixtureEnergies();
+
+ /*--- Setting Conservative Variables ---*/
+ for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+   U_inlet[iSpecies] = rho*Mass_Frac[iSpecies];
+ for (iDim = 0; iDim < nDim; iDim++)
+   U_inlet[nSpecies+iDim] = rho*Velocity[iDim];
+ U_inlet[nVar-2] = rho*(energies[0]+0.5*sqvel);
+ U_inlet[nVar-1] = rho*(energies[1]);
+
+ /*--- Setting Primitive Vaariables ---*/ 
+ for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+   V_inlet[iSpecies] = Mass_Frac[iSpecies]*rho;
+ V_inlet[nSpecies] = Temperature;
+ V_inlet[nSpecies+1] = Temperature_ve;
+ for (iDim = 0; iDim < nDim; iDim++)
+   V_inlet[nSpecies+2+iDim] = Velocity[iDim];
+ V_inlet[nSpecies+2+nDim] = Pressure;
+ V_inlet[nSpecies+3+nDim] = rho;
+ V_inlet[nSpecies+4+nDim] = U_inlet[nVar-2]+Pressure/rho;
+ V_inlet[nSpecies+5+nDim] = soundspeed;
+ V_inlet[nSpecies+6+nDim] = U_inlet[nVar-2]/Temperature;
+ V_inlet[nSpecies+7+nDim] = U_inlet[nVar-1]/Temperature_ve;
+
+ /*--- Loop over all the vertices on this boundary marker ---*/
+ for(iVertex = 0; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+   iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+   /*--- Check if the node belongs to the domain (i.e, not a halo node) ---*/
+   if (geometry->nodes->GetDomain(iPoint)) {
+
+     /*--- Index of the closest interior node ---*/
+     Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+
+     /*--- Current solution at this boundary node ---*/
+     for (iVar = 0; iVar < nVar; iVar++) U_domain[iVar] = nodes->GetSolution(iPoint,iVar);
+     for (iVar = 0; iVar < nPrimVar; iVar++) V_domain[iVar] = nodes->GetPrimitive(iPoint,iVar);
+
+     /*--- Normal vector for this vertex (negate for outward convention) ---*/
+     geometry->vertex[val_marker][iVertex]->GetNormal(Normal);
+     for (iDim = 0; iDim < nDim; iDim++) Normal[iDim] = -Normal[iDim];
+
+     su2double Area = 0.0;
+     for (iDim = 0; iDim < nDim; iDim++)
+       Area += Normal[iDim]*Normal[iDim];
+     Area = sqrt (Area);
+
+     /*--- Set various quantities in the solver class ---*/
+     conv_numerics->SetNormal(Normal);
+     conv_numerics->SetConservative(U_domain, U_inlet);
+     conv_numerics->SetPrimitive(V_domain, V_inlet);
+
+     /*--- Pass supplementary info to CNumerics ---*/ //TODO: why pass node infty?
+     conv_numerics->SetdPdU(nodes->GetdPdU(iPoint), node_infty->GetdPdU(0));
```

Comment:
This would need to be updated for the specific inlet condition.

I wonder if a temporary variable structure could be initialized in this function or in CNEMOEulerSolver if we have a supersonic inlet.  Then all this math would be done once early, but just a thought.